### PR TITLE
API-27468: Add Sidekiq `current_retry` Middleware / Retry `VAForms::FormReloader` Job

### PR DESCRIPTION
**Pull request by kristen-brown**
## Summary
This PR adds new Sidekiq middleware called `SetCurrentRetry`, which sets the worker's current retry count so that it can be read inside the worker itself. If the worker class adds the attribute accessor, `attr_accessor :current_retry`, it will have its `@current_retry` set. If the worker class does not include the attribute accessor, the assignment of `worker.current_retry` will simply be skipped.

The PR also includes enhancements to the `VAForms::FormReloader` job that allow the job to retry up to 7 times if a request to the form URL returns something other than a `200` or something other than a PDF. If the job exhausts its retries, the `valid_pdf` attribute on the `VAForms::Form` is set to `false`, which excludes the form from the [VA.gov form search](https://www.va.gov/find-forms/). This PR also adds additional/enhanced Slack alerting, Rails error logging, and a StatsD counter for total job failures. A handful of unused VCR cassettes were removed as well.

## Related issue(s)
[API-27468](https://jira.devops.va.gov/browse/API-27468)

## Testing done
I've added specs for the `SetCurrentRetry` and `VAForms::FormReloader` classes and ran the `VAForms::FormReloader` locally a handful of times. Further testing will be conducted on the development server.

## Screenshots
None

## What areas of the site does it impact?
The `SetCurrentRetry` middleware affects all Sidekiq jobs, but will only assign `worker.current_retry` if the worker has included  `attr_accessor :current_retry`, of which `VAForms::FormReloader` is the only worker to do so today. The `VAForms` changes impact the `va_forms` module only, and specifically the `VAForms::FormReloader` worker only.

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation) – N/A
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature – N/A

## Requested Feedback
Would like an extra set of eyes on the Sidekiq middleware addition, since this isn't something that I've done before.

### CODE REVIEW

File: modules/va_forms/spec/workers/form_builder_spec.rb [hunk 1]
```diff
@@ -6,72 +6,237 @@
...
+            expect(Rails.logger).to have_received(:error).with(error_message, error_details)
+          end
+        end
       end
     end
 
     it 'expands relative urls' do
       test_url = './medical/pdf/vha10-10171-fill.pdf'
-      final_url = VAForms::FormBuilder.new.expand_va_url(test_url)
+      final_url = described_class.new.expand_va_url(test_url)
```
Comment (mathisto) [inline modules/va_forms/spec/workers/form_builder_spec.rb:227]: I am a dufus and should have used this initially. Nice catch.

**Review by mathisto — DISMISSED** [review]
Love the middleware!

**Review by rmtolmach — DISMISSED** [review]
LGTM

**Review by mathisto — APPROVED** [review]
Wow. Fantastic work on the tricky specs for this.

**Review by mathisto — APPROVED** [review]

**Review by caseywilliams — APPROVED** [review]
Nice work!!

**Pull request merged**